docs/madr/decisions/080-inbound-policies-matches.md

```diff
@@ -45,26 +74,436 @@ Version in parenthesis indicates which Kuma release is going to support the user
+           - spiffeId:
+               type: Exact
+               value: "spiffe://trust-domain.mesh/ns/default/sa/api-gateway"
+        allowWithShadowDeny:
+           - spiffeId:
+               type: Prefix
+               value: "spiffe://trust-domain.mesh/ns/legacy"
+        allow:
+           - spiffeId:
+               type: Prefix
+               value: "spiffe://trust-domain.mesh/"
+```
+
+To spare the readability we'll add a syntactic sugar for a single rule without `matches`.
+For example, the following inbound policies are identical:
+
+```yaml
+type: InboundPolicy
+mesh: default
+name: typical-inbound-policy
+spec:
+  targetRef: {}
+  rules:
+     - default: {} # matches all requests
+---
+type: InboundPolicy
+mesh: default
+name: same-but-with-syntactic-sugar
+spec:
+  targetRef: {}
+  default: {} # matches all requests
+```
+
+With syntactic sugar, MeshTrafficPermission will look exactly as described in [Option 2](#option-2-meshtrafficpermission-is-a-single-item-policy)
+
+```yaml
+type: MeshTrafficPermission
+mesh: default
+name: by-service-owner
+spec:
+  targetRef:
+     kind: Dataplane
+     labels:
+       app: backend
+     sectionName: http-port
+  default:
+    deny:
+      - spiffeId:
+           type: Exact
+           value: "spiffe://trust-domain.mesh/ns/default/sa/api-gateway"
+    allowWithShadowDeny:
+       - spiffeId:
+           type: Prefix
+           value: "spiffe://trust-domain.mesh/ns/legacy"
+    allow:
+       - spiffeId:
+            type: Prefix
+            value: "spiffe://trust-domain.mesh/"
+```
+
+Evaluation rules:
+1. If the incoming request matches at least one matcher in `deny` list – the result is DENY.
+2. If the incoming request matches at least one matcher in either `allow` or `allowWithShadowDeny` list – the result is ALLOW.
+3. If the incoming request doesn't match anything – the result is DENY.
+
+#### Algorithm
+
+1. Collect all MeshTrafficPermissions that target the inbound
+2. Concat all `rules` arrays (we don't want to merge `default` confs to solve the [Problem 2](#problem-2))
+
+```yaml
+rules:
+   - default: # from 'by-mesh-operator' MTP
+       deny:
+          - spiffeId:
+              type: Exact
+              value: "spiffe://trust-domain.mesh/ns/default/sa/frontend"
+   - default: # from 'by-service-owner' MTP
+        deny:
+           - spiffeId:
+                type: Exact
+                value: "spiffe://trust-domain.mesh/ns/default/sa/api-gateway"
+        allowWithShadowDeny:
+           - spiffeId:
+                type: Prefix
+                value: "spiffe://trust-domain.mesh/ns/legacy"
+        allow:
+           - spiffeId:
+                type: Prefix
+                value: "spiffe://trust-domain.mesh/"
+```
+
+3. Starting this step MTP plugin is in charge, it receives the struct from step 2 and generates Envoy config.
+For each item in `rules` generate appropriate `matcher_list` and then concat `matcher_list` from old rules prioritizing `Deny`:
```

Comment:
> old rules prioritizing `Deny`

is this easy to do?

Reply:
I don't know what happened here, but I wanted to type "all rules" not "old rules" 🤦‍♂️